### PR TITLE
Exceptions on module init

### LINKED ISSUE #269: any exceptions raised during module initialisation should be reported in stampy-error-log
We have a channel `#stampy-error-log` where bot exceptions are reported, except for those raised during module initialisation

### PR DESCRIPTION
**Pull request by MatthewBaggins**
- Fixed reporting exceptions in stampy-error-log on processing message
- Added reporting exceptions in stampy-error-log on module initialization ( closes #269 )
- Moved `coda_api_token` to config and made Coda optional
- Fixed table id for Stampy Answers API on production
- Error fixes related to checking user roles, testing whether channels exist, etc.

### CODE REVIEW

**Review by ProducerMatt — APPROVED** [review]

File: config.py [hunk 1]
```diff
@@ -158,25 +182,25 @@ def getenv_unique_set(var_name, default=frozenset()) -> frozenset[str]:
     # private channel is where stampy logging gets printed
     bot_private_channel_id = getenv("BOT_PRIVATE_CHANNEL_ID", default=None)
     # NOTE: Rob's invite/member management functions, not ported yet
-    can_invite_role_id = getenv_unique_set("CAN_INVITE_ROLE_ID", default=None)
     member_role_id = getenv("MEMBER_ROLE_ID", default=None)
     # bot_reboot is how stampy reboots himself
-    bot_reboot = getenv("BOT_REBOOT", default=False)
+    bot_reboot = getenv("BOT_REBOOT", default=None) is not None
```
Comment (ProducerMatt) [inline config.py:187]: This is not a Boolean, please undo

**Review by ProducerMatt — CHANGES_REQUESTED** [review]

File: servicemodules/discord.py [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
+from utilities import (
+    Utilities,
+    get_question_id,
+    is_test_response,
+    is_test_message,
+    is_test_question,
+    get_git_branch_info,
+    limit_text,
+)
+from utilities.discordutils import DiscordMessage
```
[the remark this answers is not in the document]
Comment (MatthewBaggins) [reply servicemodules/discord.py:33]: First, stdlib imports, then third-party imports, then local modules. Standard Python ordering of imports

File: config.py (hunk 1 above)
Comment (MatthewBaggins) [reply config.py]: Sorry, undone. Although... is there a reason not to make it a `bool` if there are exactly two values allowed for this variable? (`"exec"` and `False`).

**Review by ProducerMatt — APPROVED** [review]

**Comment by ProducerMatt** [comment]
To answer your question @MatthewBaggins I made `BOT_REBOOT` a list of options because I had a feeling that it may be useful in the future for people who want to run Stampy in different system setups.

**Pull request merged**